### PR TITLE
Mkozel

### PR DESCRIPTION
**Pull request by Maks122525**

### CODE REVIEW

File: SRS.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+![редактировать](https://user-images.githubusercontent.com/65542353/127388771-dee34b02-ea34-49c5-9511-cdb05d5894e1.png)
+
+##### Removing the department
+___Main scenario:___ 
+* The user, while in the list of departments mode, presses the "Delete" button in the selected department line; 
+* Application displays confirmation dialog “Please confirm delete department?”; 
+* The user confirms the removal of the department; 
+* Record is deleted from database; 
+* If error occurs, then error message displays; 
+* If a department record is successfully deleted, a list of departments with no deleted records is displayed and employees with the deleted department are deleted.
```
Comment (inauros) [inline SRS.md:69]: I don't think it is a good idea to remove employees without additional confirmation. If the user deletes a department with associated employees, he/she should confirm separately deleting departments and separately deleting employees. This approach will allow avoiding accidental deletion of employees from the database in the future.

**Review by inauros — CHANGES_REQUESTED** [review]
1. Please fix **conflicting files**.
2. In accordance with the task, a folder "documentation" should be created and SRS should be placed in it.

**Comment by KisKass** [comment]
add error
edit error
delete error

File: SRS.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+___Main scenario:___ 
+* User clicks the “Edit” button in the department list view mode; 
+* Application displays form to enter department data;
+* User enters department data and presses “Save” button; 
+* If any data is entered incorrectly, incorrect data messages are displayed; 
+* If entered data is valid, then edited data is added to database; 
+* If error occurs, then error message is displaying; 
+* If the vehicle record is successfully edited, the list of departments with the added records is displayed and the data in the "employees" table is changed.
+
+___Cancel operation scenario:___
```
[the remark this answers is not in the document]
Comment (Maks122525) [reply SRS.md]: Я добавил общий сценарий в конец документации про отделы

File: SRS.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+* Name department – departament name; 
+* Average salary – Average wages of employees; 
+
+
+##### Add department
+___Main scenario:___ 
+* User clicks the “Add” button in the department list view mode; 
+* Application displays form to enter department data; 
+* User enters department data and presses “Save” button;
+* If any data is entered incorrectly, incorrect data messages are displayed;
```
[the remark this answers is not in the document]
Comment (inauros) [reply SRS.md]: I still don't see a mockup for a case when input is incorrect.

File: SRS.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+![2](https://user-images.githubusercontent.com/65542353/127388666-6bee0a67-c101-4f5c-a217-d147a66aa396.png)
+
+When adding a department, the following details are entered: 
+* Name department – departament name; 
+* Average salary – Average wages of employees; 
+
+Constraints for data validation:
+
+* Name department – maximum length of 30 characters;
```
[the remark this answers is not in the document]
Comment (inauros) [reply SRS.md]: Not fixed yet

File: SRS.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+Constraints for data validation:
+
+* Name department – maximum length of 30 characters; 
+* Average salary- unique, maximum length of 5 characters; . 
+ ##### Edit department
+___Main scenario:___ 
+* User clicks the “Edit” button in the department list view mode; 
+* Application displays form to enter department data;
+* User enters department data and presses “Save” button; 
+* If any data is entered incorrectly, incorrect data messages are displayed;
```
[the remark this answers is not in the document]
Comment (inauros) [reply SRS.md]: I still don't see a mockup for a case when input is incorrect.

File: SRS.md (hunk 1 above)
Comment (inauros) [reply SRS.md]: Not fixed yet

**Comment by inauros** [comment]
@Maks122525 - please remove the part about employees from your SRS.

File: SRS.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (Maks122525) [reply SRS.md]: fixed

File: SRS.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (Maks122525) [reply SRS.md]: fixed

File: SRS.md (hunk 1 above)
Comment (Maks122525) [reply SRS.md]: I added the item "Confirm the removal of employees of this department"

File: SRS.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (Maks122525) [reply SRS.md]: Fixed

**Review by inauros — APPROVED** [review]

**Review by KisKass — APPROVED** [review]

**Review by IlyaRozumouski — APPROVED** [review]

**Pull request merged**